### PR TITLE
adding example back / fixes from the original repo related to memory …

### PR DESCRIPTION
**Pull request by maxalencar**
- adding CI
- fixes from the original repo related to memory management

### CODE REVIEW

File: example/app.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/dojo-engineering/applepay"
+	"github.com/gin-gonic/gin"
```
Comment (tetsuo-ps) [inline example/app.go:11]: [nit] is there a reason we have gin here instead of echo?  oh wait, is this a sample app from an entirely separate source that we're just including in this repo?  🤔   Is there a source for this example?  Where did this come from?  The PR description just says "adding example back"...

File: example/decrypt/main.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
...nMy5jcmwwDgYDVR0PAQH/BAQDAgEGMBAGCiqGSIb3Y2QGAg4EAgUAMAoGCCqGSM49BAMCA2cAMGQCMDrPcoNRFpmxhvs1w1bKYr/0F+3ZD3VNoo6+8ZyBXkK3ifiY95tZn5jVQQ2PnenC/gIwMi3VRCGwowV3bF3zODuQZ/0XfCwhbZZPxnJpghJvVPh6fRuZy5sJiSFhBpkPCZIdAAAxggGLMIIBhwIBATCBhjB6MS4wLAYDVQQDDCVBcHBsZSBBcHBsaWNhdGlvbiBJbnRlZ3JhdGlvbiBDQSAtIEczMSYwJAYDVQQLDB1BcHBsZSBDZXJ0aWZpY2F0aW9uIEF1dGhvcml0eTETMBEGA1UECgwKQXBwbGUgSW5jLjELMAkGA1UEBhMCVVMCCCRD8qgGnfV3MA0GCWCGSAFlAwQCAQUAoIGVMBgGCSqGSIb3DQEJAzELBgkqhkiG9w0BBwEwHAYJKoZIhvcNAQkFMQ8XDTE3MDIwMTE4NDUwNlowKgYJKoZIhvcNAQk0MR0wGzANBglghkgBZQMEAgEFAKEKBggqhkjOPQQDAjAvBgkqhkiG9w0BCQQxIgQglCIj48n0VuU/n1ZcRRQGtOUg3PoSbmRT4t6T7AwHieswCgYIKoZIzj0EAwIERjBEAiBARLdtQbAkukYzQy2sf4RKI5fZTliIsZjzR6rhSkFJWQIgGr++I+0XiSFAxs/QRGJuMOM+UnuKQea28cwVPd1mHZsAAAAAAAA=","data":"qMvuwnxckj/BzZgR7bR75QqCB+CmEo8AYPJDZl+oD/eZgzcvHB1UfwXdyOIjQk1NX0whfwPZoh6Xfkxvb6g0F9Y/dTtJW4E5aD39NDlaZD5C8XOyDlx27IXCOc5vkEyfV4z1T15wsmYKRl0K+BcmaLbuYEmCQGTwq4Z1LVLhlDpkbtdrqr7WuBH6mIToV4AV+zlTfKj67uLRpmhqBLox14hFkVl3Il15Oq6PnYP2f+padZudUrkjWOPR8pNepPF52EL/mUNadKs3NjqG9uJLl2ELY1A+MESosJ6zoSpKuBBF8FxvaJQgDCJS2yeOut+r8okbh06xVMjNPLC7dGyGW2a6OdpsMGc5+nsP9bs3V6NIosYDCoszEBVFFFjjnYSJhdlER1i6lGE6RTjSUnPGMSb4aknjrDRN/4AJz4Q="}}`), token)
```
Comment (tetsuo-ps) [inline example/decrypt/main.go:23]: [nit] I wonder if we should just have files in a `fixtures` directory (or `resources` or something) and then load the file text into a variable for usage.  Then the actual file can be formatted in a readable way.

File: example/app.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	// h, err := r.Token.PublicKeyHash()
+
+	token, err := ap.DecryptResponse(r)
+	if err != nil {
+		log.Println(err)
+		c.Status(http.StatusBadRequest)
+		return
+	}
+
+	fmt.Println("Token received!")
```
Comment (tetsuo-ps) [inline example/app.go:92]: [nit] I see a combination of `log.Println` and `fmt.Println`, is that intentional or should some of these log statements be removed/updated?

File: session.go [hunk 4]
```diff
@@ -81,17 +81,24 @@ func (m Merchant) Session(url string) (sessionPayload []byte, err error) {
...
+	if err := json.NewEncoder(buf).Encode(m.sessionRequest()); err != nil {
+		return nil, errors.Wrap(err, "error encoding the request")
+	}
+
+	cl := m.authenticatedClient()
 	res, err := cl.Post(url, "application/json", buf)
 	if err != nil {
 		return nil, errors.Wrap(err, "error making the request")
 	}
+	defer res.Body.Close()
```
Comment (tetsuo-ps) [inline session.go:94]: 👍

File: token_signature.go [hunk 5]
```diff
@@ -276,13 +284,16 @@ func (t PKPaymentToken) extractSigningTime(p7 *C.PKCS7) (time.Time, error) {
...
+			stBio.Free()
 			return time.Time{}, errors.Wrap(opensslErr(), "time encoding error")
 		}
 		pt, err := time.Parse("Jan _2 15:04:05 2006 MST", stBio.ReadAllString())
 		if err != nil {
+			stBio.Free()
 			return time.Time{}, errors.Wrap(err, "error parsing time")
 		}
 		signingTime = pt
+		stBio.Free()
```
Comment (tetsuo-ps) [inline token_signature.go:296]: (I have to put this comment on a line that's changed, even though github actions can tag a completely unrelated code line, which is super annoying) 

`return unsafe.Pointer(uintptr(ptr))`

Is this something we should change/fix?  and/or can we add a comment about why we're using this and why we can't change it otherwise?  I suspect this error from github won't just go away...

**Review by tetsuo-ps — COMMENTED** [review]
Various questions, I think things look generally ok but we should clear up the unsafe pointer bit!

File: example/app.go (hunk 1 above)
Comment (maxalencar) [reply example/app.go:11]: so this repo is a fork of https://github.com/processout/applepay, the sample app is from there, we could do it ourselves or just remove like I did before, I just decided to put it back as I needed to do more work for the mobile sdk so I thought in case someone else would do such a job, they would like to have an example to test, so I got it back from the original repo, if you think it's confusing or dirty I can remove again no problem :)

File: example/decrypt/main.go (hunk 2 above)
Comment (maxalencar) [reply example/decrypt/main.go:23]: this is just example code for testing, I didn't bother to tidy up, I think I am just gonne remove it as it's making this PR confusing haha I just wanted to give simple examples to help people to test the library, I didn't bother with structuring nicelly or tidying up, I will remove it then ;)

File: token_signature.go (hunk 5 above)
Comment (maxalencar) [reply token_signature.go:296]: yeah that's because I added pipeline to do go vet and linter, this line got caught there, we could add a task to fix that, but to be honest I rather not try anything now as this is not my code and it's working this way, trying to fix this might take some time and effort testing and make sure that it's working in prod, I rather not take the risk at this point.

File: example/decrypt/main.go (hunk 2 above)
Comment (maxalencar) [reply example/decrypt/main.go]: done :)

File: example/app.go (hunk 3 above)
Comment (maxalencar) [reply example/app.go]: I removed the example, so this issue is not relevant anymore :)

File: example/app.go (hunk 1 above)
Comment (maxalencar) [reply example/app.go]: removed

**Review by tetsuo-ps — APPROVED** [review]
Let's create a backlog ticket to audit that `token_signature.go` code (not sure where it came from, but if we think it's safe, at the very least we should figure out how to exclude it from the linter check
